### DISCUSSION TITLE
flutter_rust_bridge_codegen build-web errors, WASM can't compile

### DISCUSSION THREAD

**Discussion by Umenokin** [fzyzcjy/flutter_rust_bridge#2236] [Q&A] [answered]
This question related to issue which is was not able to reopen: https://github.com/fzyzcjy/flutter_rust_bridge/issues/2161

## Error
```
user@MacBook my_app % flutter_rust_bridge_codegen build-web
> which wasm-pack (pwd: null, env: null)
/Users/user/.cargo/bin/wasm-pack
> cargo read-manifest (pwd: rust, env: null)
> wasm-pack build -t no-modules -d /Users/user/dev/my_app/web/pkg --no-typescript --out-name rust_lib_my_app --dev rust -- -Z build-std=std,panic_abort (pwd: null, env: {RUSTUP_TOOLCHAIN: nightly, RUSTFLAGS: -C target-feature=+atomics,+bulk-memory,+mutable-globals, CARGO_TERM_COLOR: always})
[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
error: "/Users/user/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/Cargo.lock" does not exist, unable to build with the standard library, try:
        rustup component add rust-src --toolchain nightly-aarch64-apple-darwin
Error: Compiling your crate to WebAssembly failed
Caused by: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
  full command: cd "rust" && "cargo" "build" "--lib" "--target" "wasm32-unknown-unknown" "-Z" "build-std=std,panic_abort"
Unhandled exception:
ProcessException: Bad exit code (1). If you want to see extra information, set FRB_DART_RUN_COMMAND_STDERR=1
  Command: wasm-pack build -t no-modules -d /Users/user/dev/my_app/web/pkg --no-typescript --out-name rust_lib_my_app --dev rust -- -Z build-std=std,panic_abort
#0      runCommand (package:flutter_rust_bridge/src/cli/run_command.dart:67:5)
<asynchronous suspension>
#1      _executeWasmPack (package:flutter_rust_bridge/src/cli/build_web/executor.dart:151:3)
<asynchronous suspension>
#2      executeBuildWeb (package:flutter_rust_bridge/src/cli/build_web/executor.dart:67:3)
<asynchronous suspension>
#3      BuildWebCommand.run (package:flutter_rust_bridge/src/cli/build_web/entrypoint.dart:18:5)
<asynchronous suspension>
#4      CommandRunner.runCommand (package:args/command_runner.dart:212:13)
<asynchronous suspension>
#5      main (file:///Users/user/.pub-cache/hosted/pub.dev/flutter_rust_bridge-2.2.0/bin/flutter_rust_bridge.dart:8:3)
<asynchronous suspension>
Error: Fail to execute command, please see logs above for details.
```

## Toolchain
```
user@MacBook my_app % rustup default
nightly-aarch64-apple-darwin (default)
```

## Details
I think this issue is related to Cargo issue, which was resolved yesterday but still not able to make a wasm build:
https://github.com/rust-lang/cargo/pull/14370

I also tried reinstalling rust-std but it did not help as well as creating new project :
```
user@MacBook my_app % rustup component remove rust-src --toolchain nightly-aarch64-apple-darwin
info: removing component 'rust-src'
user@MacBook my_app % rustup component add rust-src --toolchain nightly-aarch64-apple-darwin
info: downloading component 'rust-src'
info: installing component 'rust-src'
```

**Comment by fzyzcjy** [comment]
Hi, I reopened that issue and feel free to ask there!

Same as what I replied before,

> I do not have a reproducible environment, so cannot debug for this. If you like to test more, maybe clone frb repo and modify the ./frb_dart code related to building web to print out things etc to debug.

**Comment by wxitcode** [comment, marked as answer]
just update rust nightly

'''
$ rustup update nightly
'''

**Comment by Umenokin** [comment]
So now I was able to build a project with `flutter_rust_bridge_codegen build-web`
But how can I run it?

While `flutter run` works on macOS or iOS, I was never able to run it in Chrome.
`flutter run > select chrome` or `flutter run -d chrome` result in error:

<img width="561" alt="Screenshot 2024-08-11 at 17 08 40" src="https://github.com/user-attachments/assets/6ea6961d-d1b3-4188-a681-9653c8aa60cb">

**Reply by Umenokin** [reply]
I found working solution in one of here: https://scrapbox.io/hitech/Flutter%E3%81%A8Rust%E3%82%92%E7%B0%A1%E5%8D%98%E3%81%AB%E6%A7%8B%E7%AF%89%E3%81%99%E3%82%8B
`flutter run --web-header=Cross-Origin-Opener-Policy=same-origin --web-header=Cross-Origin-Embedder-Policy=credentialless`

However, every time you refresh the page, you have 50/50 of a chance it would work:
<img width="1176" alt="Screenshot 2024-08-11 at 17 20 25" src="https://github.com/user-attachments/assets/4efd516a-317c-426b-9667-f49b85b446c1">
<img width="1176" alt="Screenshot 2024-08-11 at 17 20 16" src="https://github.com/user-attachments/assets/f298069c-f9f4-4023-8216-f9815f98a435">
<img width="1176" alt="Screenshot 2024-08-11 at 17 20 07" src="https://github.com/user-attachments/assets/c91506e4-cbaa-49d9-aa98-4d46e458e7a5">

**Reply by fzyzcjy** [reply]
The quickstart doc of flutter_rust_bridge mentions to add those headers. For the new error, it may be related to standard Flutter, since it looks like loading something during bootstrap, and flutter_rust_bridge seems to has not been touched. Could you please check whether there are some Flutter discussions/solutions about this?
